actions/recurring_reservation.py

```diff
@@ -0,0 +1,185 @@
+            start_date=self.recurring_reservation.begin_date,
+            end_date=self.recurring_reservation.end_date,
+        )
+        sorted_closed = sorted(closed.get(pk, []), key=lambda x: x.buffered_start_datetime)
+        sorted_blocked = sorted(blocked.get(pk, []), key=lambda x: x.buffered_start_datetime)
+        closed_timespans = merge_overlapping_time_span_elements(sorted_closed)
+        blocked_timespans = merge_overlapping_time_span_elements(sorted_blocked)
+
+        reservable_timespans = self.get_reservable_timespans() if check_opening_hours else []
+
+        results = ReservationSeriesCalculationResults()
+
+        begin_time: datetime.time = self.recurring_reservation.begin_time
+        end_time: datetime.time = self.recurring_reservation.end_time
+
+        weekdays: list[int] = [int(val) for val in self.recurring_reservation.weekdays.split(",")]
+        if not weekdays:
+            weekdays = [self.recurring_reservation.begin_date.weekday()]
+
+        for weekday in weekdays:
+            delta: int = weekday - self.recurring_reservation.begin_date.weekday()
+            if delta < 0:
+                delta += 7
+
+            begin_date: datetime.date = self.recurring_reservation.begin_date + datetime.timedelta(days=delta)
+
+            periods = get_periods_between(
+                start_date=begin_date,
+                end_date=self.recurring_reservation.end_date,
+                start_time=begin_time,
+                end_time=end_time,
+                interval=self.recurring_reservation.recurrence_in_days,
+                tzinfo=DEFAULT_TIMEZONE,
+            )
+            for begin, end in periods:
+                if begin.date() in skip_dates:
+                    continue
+
+                reservation_timespan = TimeSpanElement(
+                    start_datetime=begin,
+                    end_datetime=end,
+                    is_reservable=True,
+                    buffer_time_after=self.recurring_reservation.reservation_unit.buffer_time_after,
+                    buffer_time_before=self.recurring_reservation.reservation_unit.buffer_time_before,
+                )
+
+                # Would the reservation timespan overlap with any closing timespans
+                # that exist due to existing reservations? Checks for:
+                # 1) Unbuffered reservation timespan overlapping with any buffered closed timespan
+                # 2) Unbuffered closed timespan overlapping with any buffered reservation timespan
+                # 3) Unbuffered reservation timespan overlapping with any unbuffered blocked timespan
+                if (
+                    any(reservation_timespan.overlaps_with(closed) for closed in closed_timespans)
+                    or any(closed.overlaps_with(reservation_timespan) for closed in closed_timespans)
+                    or any(reservation_timespan.overlaps_with(blocked) for blocked in blocked_timespans)
+                ):
```

Comment:
`closed` and `blocked` shadows the variable names declared earlier.
This is really **not an issue that must be fixed**, since the original variables aren't used later, but maybe just a bit misleading? 😅